lib/scenic/adapters/oracle.rb

```diff
@@ -76,6 +76,7 @@ def view_dependencies
           from user_objects uo
             left join user_dependencies ud on uo.object_name = ud.name
               and ud.referenced_type in ('VIEW', 'MATERIALIZED VIEW')
+              and ud.object_name in (select object_name from user_objects)
```

Comment:
I think you meant `ud.referenced_name` here:
```suggestion
              and ud.referenced_name in (select object_name from user_objects)
```

I've also added a spec to exercise this in main. Could you rebase that into this branch so we can verify that it fixes the issue?